hasura/graphql-engine
# console: allow entering big int values in the console

* convert bigint to string before JSON parse
* prevent auto round off and shows accurate value




### Description

v8 has auto round off to nearest safe value, this will appear as a bug in the UI. 
by converting the big int values to string, UI will render the values correctly.


### Changelog


### Affected components



### Related Issues

close #3667 



### Solution and Design

convert bigint to string before JSON parse.


### Steps to test and verify


1. add a table in the UI with column with bigint type.
2. go to graphiql and insert a bigint value not ending with zeros (with zeros we cannot test the round of scenario)
3. go-to data table and verify the big int value is displayed correctly.

### Limitations, known bugs & workarounds


* Considering the solution involves a conditional type change, any arithmetic operations without pre-type check might get affected. 

### Server checklist


#### Catalog upgrade

Does this PR change Hasura Catalog version?

#### Metadata


Does this PR add a new Metadata feature?
  - Does `run_sql` auto manages the new metadata through schema diffing?
  - Does `run_sql` auto manages the definitions of metadata on renaming?
  - Does `export_metadata`/`replace_metadata` supports the new metadata added?


#### GraphQL
   
   

#### Breaking changes


  1. Metadata API

     Existing `query` types:
     

  2. GraphQL API

     Schema Generation:
     
     
     

     Schema Resolve:-
     
     

  3. Logging
